fireant/tests/queries/test_data_blending_integration.py

```diff
@@ -305,22 +300,104 @@ def test_include_mapped_field_in_subqueries_when_the_aliases_are_different(self)
             '"a" "$a",'
             '"metric" "$metric0" '
             'FROM "test0" '
-            'GROUP BY "$a" '
-            'ORDER BY "$a"'
+            'GROUP BY "$a"'
             ') "sq0" '
-            "JOIN ("
+            "LEFT JOIN ("
             "SELECT "
             '"b" "$b",'
             '"metric" "$metric1" '
             'FROM "test1" '
-            'GROUP BY "$b" '
-            'ORDER BY "$b"'
+            'GROUP BY "$b"'
             ') "sq1" ON "sq0"."$a"="sq1"."$b" '
             'ORDER BY "$a"',
             str(query),
         )
 
 
+class MultipleDatasetsBlendedEdgeCaseTests(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        db = Database()
+        t0, t1, t2 = Tables("test0", "test1", "test2")
+        cls.primary_ds = DataSet(
+            table=t0,
+            database=db,
+            fields=[
+                Field(
+                    "timestamp",
+                    label="Timestamp",
+                    definition=t0.timestamp,
+                    data_type=DataType.date,
+                ),
+                Field(
+                    "metric0",
+                    label="Metric0",
+                    definition=t0.metric,
+                    data_type=DataType.number,
+                ),
+            ],
+        )
+        cls.primary_ds.id = 0
+        cls.secondary_ds = DataSet(
+            table=t1,
+            database=db,
+            fields=[
+                Field(
+                    "timestamp",
+                    label="Timestamp",
+                    definition=t1.timestamp,
+                    data_type=DataType.date,
+                ),
+                Field(
+                    "metric1",
+                    label="Metric1",
+                    definition=t1.metric,
+                    data_type=DataType.number,
+                ),
+            ],
+        )
+        cls.secondary_ds.id = 1
+        cls.tertiary_ds = DataSet(
+            table=t2,
+            database=db,
+            fields=[
+                Field(
+                    "timestamp",
+                    label="Timestamp",
+                    definition=t2.timestamp,
+                    data_type=DataType.date,
+                ),
+                Field(
+                    "metric2",
+                    label="Metric2",
+                    definition=t2.metric,
+                    data_type=DataType.number,
+                ),
+            ],
+        )
+        cls.tertiary_ds.id = 2
+        cls.blend_ds = (
+            cls.primary_ds.blend(cls.secondary_ds).on_dimensions().blend(cls.tertiary_ds).on_dimensions()
+        )
+
+    def test_selecting_just_one_metric_in_non_primary_dataset(self):
```

Comment:
Is there only a need for a single new test?

Reply:
No, I have 2 more cases I'm covering before I will merge. One related to the share operation and one where no queries need to be blended.

There's also still one bug active related to references and percentage that will also introduce another testing path.

Reply:
(all of these remarks are solved now btw)